src/commands/mobile_app/upload.rs

```diff
@@ -409,7 +420,10 @@ fn upload_file(
         sha,
         build_configuration,
     )?;
-    Ok(())
+
+    response
+        .artifact_id
+        .ok_or(anyhow!("Missing artifactId in response"))
```

Comment:
The disadvantage here is that because we require the artifact ID to be present in the response, we will always have to wait until the artifact has finished assembling, which could significantly slow down the command. It looks like we are already doing this, but in the future we may want to change it, as it is likely not the ideal behavior.

If we want to print the URL, there's probably no way around waiting for the assembly to finish. But, it is likely worth reconsidering whether we really think it is worth waiting for assembly to finish, just to be able to print the URL (with no way to opt in or out of this behavior)

Reply:
Also, is the `artifact_id` always expected to be there if we hit this line (i.e., if the assembly status is OK)?

If not, I would suggest improving the error message to make it more actionable to the user.

If we do expect the `artifact_id` to be there, I would suggest we rewrite the error message to indicate that this is an internal error with the CLI and/or with the server, as it is not fixable by the user. You can do this by using `expect` rather than returning an error, as `expect` panics, and Sentry CLI's panic handler prints a message stating that an internal error has occurred, and that users should file a bug report

Reply:
Edit: Not sure the above is accurate, I am currently reviewing #2683 and https://github.com/getsentry/sentry/pull/97435 to verify this

Reply:
We've updated the way this works so now the artifact id/url is available immediately.